amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -170,9 +181,26 @@ protected Scheduler(@NonNull final CheckpointConfig checkpointConfig,
         this.retrievalConfig = retrievalConfig;
 
         this.applicationName = this.coordinatorConfig.applicationName();
+        final MultiStreamTracker multiStreamTracker = this.retrievalConfig.multiStreamTracker();
+        if(multiStreamTracker == null) {
+            final StreamConfig streamConfig = new StreamConfig(this.retrievalConfig.streamName(),
+                    this.retrievalConfig.initialPositionInStreamExtended());
+            this.appStreamTracker = Either.right(streamConfig);
```

Comment:
There is an expectation that configuration is setup correctly, for multi-stream vs single-stream. let'e validate that with null check, both in here, and also in else case.

Reply:
The current idea is customer can choose to override their current stream & initialposition with streamtracker. We can make this explicit by allowing them to set either stream+initialposition or streamtracker and then add validations.